src/HealthGPS/energy_balance_hierarchical_model.cpp

```diff
@@ -37,15 +37,30 @@ void EnergyBalanceHierarchicalModel::update_risk_factors(RuntimeContext &context
             current_risk_factors.at(age_key) = model_age;
         }
 
-        auto &equations = definition_.get().at(model_age);
+        auto &eqns = equations_at(model_age);
         if (entity.gender == core::Gender::male) {
-            update_risk_factors_exposure(context, entity, current_risk_factors, equations.male);
+            update_risk_factors_exposure(context, entity, current_risk_factors, eqns.male);
         } else {
-            update_risk_factors_exposure(context, entity, current_risk_factors, equations.female);
+            update_risk_factors_exposure(context, entity, current_risk_factors, eqns.female);
         }
     }
 }
 
+const AgeGroupGenderEquation &EnergyBalanceHierarchicalModel::equations_at(const int &age) const {
+    auto &all_eqns = equations_.get();
+    for (auto &entry : all_eqns) {
+        if (entry.first.contains(age)) {
+            return entry.second;
+        }
+    }
+
+    if (age < all_eqns.begin()->first.lower()) {
+        return all_eqns.begin()->second;
+    }
+
+    return all_eqns.rbegin()->second;
```

Comment:
This, I believe, is to handle the case where there is no exact equation, but age is higher than the provided equation age ranges, in which case it will return the equation for the highest age range.

Reply:
I see. Maybe it needs a comment to that effect? Though commenting the code is maybe a separate endeavour...

What I meant though was that if `all_eqns` is empty, then you'll get a segfault here. There should probably be another check, either in this function or in the constructor. And maybe the same for `_variables`?

Reply:
Will make sure they're checked in the Definition constructors.